vendor/k8s.io/kubernetes/pkg/registry/deployment/etcd/etcd.go

```diff
@@ -215,6 +242,9 @@ func (r *ScaleREST) Update(ctx api.Context, name string, objInfo rest.UpdatedObj
 	if err != nil {
 		return nil, false, errors.NewNotFound(extensions.Resource("deployments/scale"), name)
 	}
+	if originalKind, exists := deployment.Annotations["original-kind.kubernetes.io"]; exists && originalKind == "Deployment.extensions" {
```

Comment:
> use the const from v1

I'm still wiring it all up and writing tests before scrubbing like that.  I'll remove the WIP when this is ready for a scrub like that.